warrior/WarriorCore/onerror_driver.py

```diff
@@ -82,44 +82,88 @@ def getErrorHandlingParameters(node, def_on_error_action, def_on_error_value, ex
         action = def_on_error_action
 
     elif action is not None and action is not False:
-        supported_values = ['next', 'goto', 'abort', 'abort_as_error']
+        supported_values = ['next', 'goto', 'abort', 'abort_as_error', "execute_and_resume"]
         action = str(action).strip()
         if action.lower() not in supported_values:
-            print_warning("unsupported option '{0}' provided for onError action, supported values are {1}".format(action, supported_values))
+            print_warning("unsupported option '{0}' provided for onError action, supported "
+                          "values are {1}".format(action, supported_values))
             print_info("Hence using default_onError action")
             action = def_on_error_action
 
     if value is None or value is False:
-        value = def_on_error_value
+        if action == "execute_and_resume":
+            print_warning("No step numbers given to go to for execute_and_resume")
+            print_info("Hence using default_onError action")
+            action = def_on_error_action
+        else:
+            value = def_on_error_value
+    else:
+        if action == "execute_and_resume":
+            value = [int(x.strip())-1 for x in value.split(",")]
 
     return action, value
 
-def next(action, value, error_handle):
+
+def next(action, value, error_handle, skip_invoked=True, print_w=True):
     """returns 'NEXT' for on_error action = next """
 
-    print_info("failure action= next")
-    error_handle['action'] = 'NEXT'
+    if skip_invoked:
+        print_info("failure action= next")
+        error_handle['action'] = 'NEXT'
+    else:
+        if print_w:
+            print_warning("Overriding onError '{0}' since this is an Invoked Step.".format('next'))
+        error_handle['action'] = 'NEXT'
     return error_handle
 
-def goto(action, value, error_handle):
-    """returns goto_step_num for on_error action = goto """
 
-    print_info("failed: failure action= goto  %s" % value)
-    error_handle['action'] = 'GOTO'
-    error_handle['value'] = value
+def goto(action, value, error_handle, skip_invoked=True):
+    """returns goto_step_num for on_error action = goto """
+    if skip_invoked:
+        print_info("failed: failure action= goto  %s" % value)
+        error_handle['action'] = 'GOTO'
+        error_handle['value'] = value
+    else:
```

Comment:
lines 127 to 129 are repeated in multiple places, can we make it a function call

Reply:
Updated.